test/one_time_password_test.rb

```diff
@@ -9,16 +9,30 @@ def setup
     @visitor = Visitor.new
     @visitor.email = 'roberto@heapsource.com'
     @visitor.run_callbacks :create
+
+    @member = Member.new
+    @member.email = nil
+    @member.run_callbacks :create
   end
 
   def test_authenticate_with_otp
     code = @user.otp_code
-
     assert @user.authenticate_otp(code)
 
     code = @visitor.otp_code
     assert @visitor.authenticate_otp(code)
   end
+  
+  def test_counter_based_otp
```

Comment:
Assignment Branch Condition size for test_counter_based_otp is too high. [17.12/15]